earth_enterprise/src/fusion/autoingest/plugins.src/Database.src

```diff
@@ -228,17 +228,15 @@ void DatabaseAssetImplD::ConstructInputsFromRequest(
                             std::vector<SharedString>* inputs) {
   // make sure the project asset_ref_string's are valid
   if (!req.config.imageryProject.empty()) {
-    RasterProjectFactory::ValidateRefForInput(req.config.imageryProject,
-                                              AssetDefs::Imagery);
+    ValidateRefForInput<MutableDatabaseAssetD, MutableDatabaseAssetVersionD>(req.config.imageryProject, AssetDefs::Imagery, kDatabaseSubtype);
```

Comment:
There's still an issue here. Even though this call is inside `MutableDatabaseAssetD`, it's calling `RasterProjectFactory`. So the call should be something like `ValidateRefForInput<MutableRasterProjectAssetD, MutableRasterProjectAssetVersionD>`. Same thing with the other calls.